dbt/compilation.py

```diff
@@ -48,43 +48,39 @@ def print_compile_stats(stats):
     logger.info("Found {}".format(stat_line))
 
 
-def prepend_ctes(model, flat_graph):
-    model, _, flat_graph = recursively_prepend_ctes(model, flat_graph)
+def prepend_ctes(model, manifest):
+    model, _, manifest = recursively_prepend_ctes(model, manifest)
 
-    return (model, flat_graph)
+    return (model, manifest)
 
 
-def recursively_prepend_ctes(model, flat_graph):
+def recursively_prepend_ctes(model, manifest):
     if dbt.flags.STRICT_MODE:
-        dbt.contracts.graph.compiled.CompiledNode(**model)
-        dbt.contracts.graph.compiled.CompiledGraph(**flat_graph)
+        # ensure that all the nodes in this manifest are compiled
+        CompiledGraph(**manifest.to_flat_graph())
 
-    model = model.copy()
-    prepend_ctes = OrderedDict()
+    prepended_ctes = OrderedDict()
 
-    if model.get('all_ctes_injected') is True:
-        return (model, model.get('extra_ctes').keys(), flat_graph)
```

Comment:
oh boy. was this ever reached before?

in either case, the thought here was that if `extra_ctes_injected` was already set, you could early exit the recursion (otherwise, we'd end up re-visiting parts of hte tree that we already visited & injected). that check is still useful here

Reply:
I don't see how this was reachable since nothing ever set `all_ctes_injected`.